master/internal/trial.go

```diff
@@ -709,6 +709,11 @@ func (t *trial) processCompletedWorkload(ctx *actor.Context, msg workload.Comple
 		ctx.Tell(ctx.Self().Parent(), sendNextWorkload{runID: t.RunID})
 		return nil
 	default:
+		if err := t.tellWithSnapshot(ctx, ctx.Self().Parent(), func(s trialSnapshot) interface{} {
```

Comment:
Is this triggering a searcher snapshot for _every_ workload?  Wasn't there a reason that we could only snapshot trials when the trial checkpointed?

Reply:
Either should work just fine (every workload was exactly what we did originally), I'll change it to only on checkpoints though, that's smarter.

Reply:
I just pushed a change to only snapshot on checkpoint; it shouldn't have a correctness difference (because of t.reset()) but may be more performant.